libraries/AP_Common/AP_Common.h

```diff
@@ -53,6 +52,20 @@
 #  define FALLTHROUGH
 #endif
 
+#ifdef __GNUC__
+ #define WARN_IF_UNUSED __attribute__ ((warn_unused_result))
+#else
+ #define WARN_IF_UNUSED
+#endif
+
+// use this to avoid issues between C++11 with NuttX and C++10 on
+// other platforms.
+#if !(defined(__GXX_EXPERIMENTAL_CXX0X__) || __cplusplus >= 201103L)
+# define constexpr const
```

Comment:
I know you don't like removing stuff on this type of PRs but can we please remove this? It's clearly not doing anything anymore (we're always building for C++11) and being in AP_Common it's even less useful (much less places include AP_Common than the HAL).